synapse/storage/databases/main/event_push_actions.py

```diff
@@ -191,7 +191,7 @@ def _get_unread_counts_by_receipt_txn(
         stream_ordering = None
 
         if last_read_event_id is not None:
-            stream_ordering = self.get_stream_id_for_event_txn(  # type: ignore[attr-defined]
+            stream_ordering = self.get_stream_id_for_event_txn(
```

Comment:
As I mentioned in #12437, for this to have any chance of working we need to change the class `EventPushActionsWorkerStore` to inherit from `StreamWorkerStore`: see my point 2 there.

Note that this may end up causing runtime failures if we run into MRO problems. If that's the case, CI will detect this.